packages/contracts-ops/contracts/test/Reboot.t.sol

```diff
@@ -4,21 +4,71 @@ pragma abicoder v2;
 
 import "forge-std/Test.sol";
 import {RebootLogic} from "../scripts/Reboot.s.sol";
+import {ReplicaTest} from "@nomad-xyz/contracts-core/contracts/test/Replica.t.sol";
+import {ReplicaHarness} from "@nomad-xyz/contracts-core/contracts/test/harnesses/ReplicaHarness.sol";
 
-contract RebootTest is RebootLogic, Test {
-    function setUp() public {}
+contract RebootTest is RebootLogic, ReplicaTest {
+    string remote;
 
-    function test_setup() public {
+    function setUp() public override {
         string memory _domain = "ethereum";
         string memory _configName = "config.json";
         __Config_initialize(_configName);
         __CallBatch_initialize(_domain, getDomainNumber(_domain), "", true);
+        // call base setup
+        super.setUp();
+        // set fake updater for ethereum & 1 remote chain
+        // before updater rotation, so it will be rotated on-chain
+        vm.writeJson(
+            vm.toString(updaterAddr),
+            outputPath,
+            protocolAttributePath(remote, "updater")
+        );
+        reloadConfig();
         // perform reboot actions
         reboot(_domain);
         // execute governance actions via vm.prank
         prankExecuteRecoveryManager(
             address(getGovernanceRouter(_domain)),
             getDomainNumber(_domain)
         );
+        // set remote domain to actual values
+        remote = getConnections(_domain)[0];
+        remoteDomain = getDomainNumber(remote);
+        // set home domain to actual values
+        homeDomain = getDomainNumber(_domain);
+        // setup replica to actual value
+        replica = ReplicaHarness(
+            address(getReplicaOf(localDomainName, remote))
+        );
+        // set committed root and optimistic timeout
+        committedRoot = replica.committedRoot();
+        optimisticTimeout = replica.optimisticSeconds();
+        // setup fake app handlers
+        setUpBadHandlers();
+        // Upgrade Replica to Replica Harness
+        // deploy ReplicaHarness implementation
+        ReplicaHarness harnessImpl = new ReplicaHarness(
```

Comment:
Okay, now I see. 

I really think that fork tests should be **identical** to the on-chain reality, with minimum changes. I think it's much better to do vm.load() and change a single storage slot than upgrading Replica to a harness.

Reply:
I can do that in a later PR.